plugins/analysis/kotlin-plugin/src/main/kotlin/arrow/meta/plugins/analysis/phases/ir/ConstraintsAnnotations.kt

```diff
@@ -95,8 +96,20 @@ private fun IrUtils.lawSubjectAnnotationFromClassId(classId: ClassId, descriptor
 
 private fun IrUtils.lawSubjectAnnotation(fnDescriptor: SimpleFunctionDescriptor, descriptor: ClassDescriptor): IrConstructorCall? =
   descriptor.irConstructorCall()?.also {
-    it.putValueArgument(0, constantValue(fnDescriptor.fqNameSafe.asString()))
+    it.putValueArgument(0, constantValue(fnDescriptor.getLawName()))
+  }
+
+private fun DeclarationDescriptor.getLawName(): String {
+  val containing = containingDeclaration
+  return when {
+    this is org.jetbrains.kotlin.descriptors.PackageFragmentDescriptor ||
+      this is org.jetbrains.kotlin.descriptors.PackageViewDescriptor ||
+      this is org.jetbrains.kotlin.descriptors.ModuleDescriptor ||
+      containing == null
+    -> fqNameSafe.asString()
+    else -> "${containing.getLawName()}/${name.asString()}"
```

Comment:
If this is calling itself we may be able to make this function `tailrec`

Reply:
Done in a new commit